generic/omemoplugin/src/omemoplugin.cpp

```diff
@@ -330,6 +358,22 @@ bool OMEMOPlugin::encryptMessageElement(int account, QDomElement &message)
         return false;
     }
 
+    // logging functionality for OMEMO-encrypted groupchats
+    if (message.attribute("type") == "groupchat") {
+        QString     room = message.attribute("to");
+        QString     from = m_mucNicks[room];
```

Comment:
```
m_mucNicks.value(room)
```
will avoid invalid empty nickname insertion if room is not there. 
Also it may be necessary to do some initialization if it's not there.

Reply:
Okay, but wouldn't .value() also return an empty string if the room is not there (unless the second parameter of `defaultValue` is specified, but I'm not sure what kind of default value would be good for this)? Qt docs say so, although I might be misunderstanding it.

Reply:
It will return empty string. that's correct. 
Since nickname can't be empty, you can check for emptiness and maybe do some initialization if empty.

Imagine a case where you enable OMEMO plugin already after joining. So you missed the initial presence and have to live with this somehow. (drop everything encrypted / force rejoin / ask user what to do)

Reply:
Okay, so I changed the behavior to logging the full jid instead of the nick in case the nick is empty. Would that be fine for now? Can't think of any elegant solution to the problem yet

Reply:
yep. should be good for now.
but technically all the MUC info has to be available somewhere in Psi including nicks.

Reply:
Well, there is a `mucNicks()` function in ContactInfoAccessingHost, but that just gives a list of nicks from a MUC (without the real jids), so I suppose the `realJid()` function could be ran on each of the nicks until the one associated with the needed account is found... But that would probably be slow, especially if the MUC is big (although I strongly doubt that there exist big OMEMO-encrypted MUCs, as that's very uncomfortable since everyone seems to have to have everyone else in their roster to encrypt all messages). Not sure what other options are there to fix this.

Reply:
It's fine to modify plugin interfaces if anything is needed.
We didn't release any stable plugins sdk yet. So it's Ok to change it.
I think after the coming release we can think about some stabilization of the plugins API.

Reply:
Oh, okay. I'll see what can be done, would be great to remove the presence tracking hack as a whole

Reply:
Thanks!